software/cm/ansible/src/main/java/org/apache/brooklyn/entity/cm/ansible/AnsiblePlaybookTasks.java

```diff
@@ -105,5 +107,27 @@ private static String optionalExtraVarsParameter(Object extraVars) {
         }
         return " --extra-vars \"@" + EXTRA_VARS_FILENAME + "\" ";
     }
+
+    public static TaskFactory<?> setUpHostsFile(String installDir, boolean force) {
+        final String hostsFile = "/etc/ansible/hosts";
+        final String tempFile = "/tmp/hosts";
+        String checkInstalled = !force ? "which ansible || " : "";
+        return sshCommands(
+                checkInstalled + " {",
+                sudo("rm -f " + hostsFile),
+                "echo 'localhost ansible_connection=local' > " + tempFile,
```

Comment:
my favourite is

    echo XXX | sudo tee /etc/privileged_file

or in java

    "echo XXX | "+sudo("tee /etc/privileged_file")

with `tee -a` to append